components/plugin_marketplace/marketplace_modal.tsx

```diff
@@ -84,81 +90,89 @@ export const InstalledPlugins = ({installedPlugins, changeTab}) => {
     return <MarketplaceList plugins={installedPlugins}/>;
 };
 
-InstalledPlugins.propTypes = {
-    installedPlugins: PropTypes.array.isRequired,
-    changeTab: PropTypes.func,
+export type MarketplaceModalProps = {
+    show: boolean;
+    plugins: MarketplacePlugin[];
+    installedPlugins: MarketplacePlugin[];
+    siteURL: string;
+    pluginStatuses?: Dictionary<PluginStatusRedux>;
+    actions: {
+        closeModal: () => void;
+        fetchPlugins: (localOnly?: boolean) => Promise<{error?: Error}>;
+        filterPlugins(filter: string): Promise<{error?: Error}>;
+    };
+};
+
+type MarketplaceModalState = {
+    tabKey: any;
+    loading: boolean;
+    serverError?: Error;
+    filter: string;
 };
 
 // MarketplaceModal is the plugin marketplace.
-export class MarketplaceModal extends React.PureComponent {
-    static propTypes = {
-        show: PropTypes.bool,
-        plugins: PropTypes.array.isRequired,
-        installedPlugins: PropTypes.array.isRequired,
-        siteURL: PropTypes.string.isRequired,
-        pluginStatuses: PropTypes.object.isRequired,
-        actions: PropTypes.shape({
-            closeModal: PropTypes.func.isRequired,
-            fetchPlugins: PropTypes.func.isRequired,
-            filterPlugins: PropTypes.func.isRequired,
-        }).isRequired,
-    }
+export class MarketplaceModal extends React.PureComponent<MarketplaceModalProps, MarketplaceModalState> {
+    private filterRef: React.RefObject<QuickInput>;
 
-    constructor(props) {
+    constructor(props: MarketplaceModalProps) {
         super(props);
 
         this.state = {
             tabKey: MarketplaceTabs.ALL_PLUGINS,
             loading: true,
-            serverError: null,
+            serverError: undefined,
             filter: '',
         };
+
+        this.filterRef = React.createRef();
     }
 
-    componentDidMount() {
+    componentDidMount(): void {
         trackEvent('plugins', 'ui_marketplace_opened');
 
         this.fetchPlugins();
 
-        if (this.refs.filter) {
-            this.refs.filter.focus();
-        }
+        this.filterRef.current?.focus();
     }
 
-    componentDidUpdate(prevProps) {
+    componentDidUpdate(prevProps: MarketplaceModalProps): void {
         // Automatically refresh the component when a plugin is installed or uninstalled.
         if (this.props.pluginStatuses !== prevProps.pluginStatuses) {
             this.fetchPlugins();
         }
     }
 
-    fetchPlugins = async () => {
+    fetchPlugins = async (): Promise<void> => {
         const {error} = await this.props.actions.fetchPlugins();
 
         this.setState({loading: false, serverError: error});
     }
 
-    close = () => {
+    close = (): void => {
         trackEvent('plugins', 'ui_marketplace_closed');
         this.props.actions.closeModal();
     }
 
-    changeTab = (tabKey) => {
+    changeTab: SelectCallback = (tabKey: any): void => {
```

Comment:
Why was `SelectCallback` added here?

Reply:
To make clear that this function has to fulfill the `SelectCallback` interface from `react-bootstrap`.